geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionSingleHopDUnitTest.java

```diff
@@ -868,585 +916,338 @@ public void testMetadataIsSameOnAllServersAndClientsHA() {
   public void testClientMetadataForPersistentPrs() throws Exception {
     LATCH.set(new CountDownLatch(4));
 
-    int port0 = vm0.invoke(() -> createPersistentPrsAndServer());
-    int port1 = vm1.invoke(() -> createPersistentPrsAndServer());
-    int port2 = vm2.invoke(() -> createPersistentPrsAndServer());
-    int port3 = vm3.invoke(() -> createPersistentPrsAndServer());
+    int locatorPort = DUnitEnv.get().getLocatorPort();
```

Comment:
dunitrunner revealed flaky failures in testClientMetadataForPersistentPrs caused by underlying BindExceptions.

The fix involves pointing the client at the servers via the locator and then let the servers use whatever ports they can find instead of restarting the servers with a previously used port.